src/nvim/screen.c

```diff
@@ -449,7 +457,16 @@ void update_screen(int type)
    */
   did_one = FALSE;
   search_hl.rm.regprog = NULL;
+
+
   FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
+    if (wp->w_redr_type == CLEAR && wp->w_floating && wp->w_grid.chars) {
+      // TODO(bfredl): this is overly expensive. The solution is to CLEAR less
```

Comment:
could add a field to `g_stats` so we have a counter of CLEARs which could be (eventually) tested.

Reply:
It could be this TODO was from before the clearmaggedon. Now basically only resize and user intervention should lead to a clear.